x/protocolpool/keeper/keeper.go

```diff
@@ -188,13 +188,13 @@ func (k Keeper) calculateClaimableFunds(ctx context.Context, recipient sdk.AccAd
 	return amount, nil
 }
 
-func (k Keeper) validateAndUpdateBudgetProposal(ctx context.Context, bp types.MsgSubmitBudgetProposal) error {
+func (k Keeper) validateAndUpdateBudgetProposal(ctx context.Context, bp *types.MsgSubmitBudgetProposal) (*types.Budget, error) {
```

Comment:
imho, it should be one or the other. Take a poiner or return a budget.

Reply:
My bad, as you pointed on slack, the arg and return value are different. You can forget this comment then 👍🏾